corehq/apps/hqwebapp/oauth_scopes.py

```diff
@@ -0,0 +1,86 @@
+import json
+import os
+from collections import defaultdict
+
+from oauth2_provider.scopes import BaseScopes
+
+from corehq.apps.domain.utils import get_domain_from_url
+from corehq.motech.fhir.const import FHIR_VERSION_4_0_1
+from corehq.motech.fhir.models import FHIRResourceType, get_schema_dir
+
+DEFAULT_SMART_SCOPES = {
+    'launch/patient': 'Access your unique patient ID',
+    'offline_access': 'Maintain access to your data when you are offline',
+    'online_access': 'Maintain access to your data when you are online',
+}
+
+HQ_DEFINED_SCOPES = {
+    'access_apis': 'Access API data on all your CommCare projects',
+}
+
+
+class FhirScopeBuilder():
+    CLIENT_TYPES = [('patient', 'your patients\''), ('user', 'your')]
+    ACCESS_PERMISSIONS = [('read', 'Read'), ('write', 'Write'), ('*', 'Read and write')]
+
+    def __init__(self, fhir_version=FHIR_VERSION_4_0_1):
+        self.fhir_version = fhir_version
+
+        self.fhir_scopes = {**DEFAULT_SMART_SCOPES}
+        self.fhir_scopes_by_resource_name = defaultdict(list)
+
+        self._build_scope_messages()
```

Comment:
nit: any chance we should execute this lazily so it doesn't add a file system read to bootstrap? (via https://github.com/dimagi/commcare-hq/pull/29504/commits/57c1c408386f3583f0cdc8d1a5e9530763d76162#diff-274fe865d204fc6c63bbdb8e0dc68f9d981aa94f72fd903b00bf94d301bd36b2R47)